enhancements/sig-policy/122-expanded-hub-templates-access/README.md

```diff
@@ -0,0 +1,310 @@
+for cluster-scoped objects or objects that have side effects are more complicated.
+
+### Goals
+
+1. A user can specify a service account for hub templates to access objects outside of the root policy namespace.
+1. The current event-driven nature of hub templates is retained with a custom service account.
+1. Tokens for the service account have an expiration of 2 hours.
+
+### Non-Goals
+
+1. Allowing service accounts outside of the root policy namespace.
+
+## Proposal
+
+### Design
+
+Define a new field at `spec.hubTemplateOptions.serviceAccountName` to specify a service account that exists in the root
+policy namespace. When specified, the service account is used instead of the service account of the Governance Policy
+Propagator for hub template execution. Additionally, no namespace or cluster-scoped restrictions will be enforced by the
+Governance Policy Propagator, and instead, it will rely on the Kubernetes API performing authorization checks of the
+service account. Note that due to the event-driven nature of hub templates, the service account must have `list` and
+`watch` permissions for the objects that the hub templates reference.
+
+Note that for backwards compatibility, if the namespace is not specified in a hub template object lookup, the root
+policy namespace will be the default, however, the service account must still have the appropriate permissions in the
+root policy namespace to perform the lookup.
+
+See an example of such a policy below:
+
+```yaml
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  name: test-config
+  namespace: test-policies
+spec:
+  disabled: false
+  remediationAction: enforce
+  hubTemplateOptions:
+    serviceAccountName: testsa
+  policy-templates:
+    - objectDefinition:
+        apiVersion: policy.open-cluster-management.io/v1
+        kind: ConfigurationPolicy
+        metadata:
+          name: test-config
+        spec:
+          remediationAction: enforce
+          severity: high
+          object-templates:
+            - complianceType: musthave
+              objectDefinition:
+                apiVersion: v1
+                data:
+                  field1: '{{hub fromConfigMap "some-other-namespace" "test-configmap" "test-field1" hub}}'
+                  field2: '{{hub fromConfigMap "some-other-namespace" "test-configmap" "test-field1" hub}}'
+                kind: ConfigMap
+                metadata:
+                  name: my-configmap
+                  namespace: test
+```
+
+### User Stories
+
+#### Story 1
+
+As a policy user, I would like to specify hub templates that reference objects outside of the root policy namespace so
+that I can stop defining additional policies on `local-cluster` to sync the required information to the root policy
+namespace.
+
+### Implementation Details/Notes/Constraints
+
+Although the user experience is rather simple, there's complexity in the implementation. One aspect is that the number
```

Comment:
Are you going to share this in tech share? Hard to understand 100% sorry too complex

Reply:
Sure, I can also do that during the daily scrum next week. If you have specific questions, I can answer those.